src/components/playback/playbackmanager.js

```diff
@@ -279,17 +279,79 @@ define(['events', 'datetime', 'appSettings', 'itemHelper', 'pluginManager', 'pla
 
     function getIntros(firstItem, apiClient, options) {
 
-        if (options.startPositionTicks || options.startIndex || options.fullscreen === false || !enableIntros(firstItem) || !userSettings.enableCinemaMode()) {
+        var unwatchedOnly = userSettings.enableCinemaTrailersUnseenOnly();
+        var trailerCount = userSettings.cinemaModeTrailerCount();
+        var enableCinema = userSettings.enableCinemaMode();
+
+        if (options.startPositionTicks || options.startIndex || options.fullscreen === false || !enableIntros(firstItem) || !enableCinema || trailerCount == 0) {
             return Promise.resolve({
                 Items: []
             });
         }
 
-        return apiClient.getIntros(firstItem.Id).then(function (result) {
+        var fetchOptions = {
+            Fields: 'RemoteTrailers,LocalTrailerCount,ServerID',
+            Filters: unwatchedOnly ? 'IsUnPlayed' : '',
+            HasTrailer: true,
+            IncludeItemTypes: 'Movie',
+            Limit: trailerCount,
+            Recursive: true,
+            SortBy: 'Random'
```

Comment:
I don't think we should be shuffling stuff server-side, we better do it client-side for server performance

Reply:
ref https://github.com/jellyfin/jellyfin-web/pull/835#discussion_r399773115